ext/intl/common/common_date.cpp

```diff
@@ -137,7 +139,8 @@ U_CFUNC int intl_datetime_decompose(zval *z, double *millis, TimeZone **tz,
 			return FAILURE;
 		}
 
-		*millis = U_MILLIS_PER_SECOND * (double)Z_LVAL(retval);
+		datetime = Z_PHPDATE_P(z);
+		*millis = U_MILLIS_PER_SECOND * ((double)Z_LVAL(retval) + datetime->time->f);
```

Comment:
@derickr I think the idea here might have been (I don't know) that people might extend `DateTime::getTimestamp()` in userland. If this is not something we want to respect, then directly fetching the value from the underlying structure is of course much better.